build.sbt

```diff
@@ -211,6 +212,43 @@ lazy val testStdLayout = Project("daffodil-test-stdLayout", file("test-stdLayout
   .dependsOn(tdmlProc % "test")
   .settings(commonSettings, nopublish)
 
+/* Workaround: certain reflection (used by JAXB) isn't allowed by default in JDK 17:
+ * https://docs.oracle.com/en/java/javase/17/migrate/migrating-jdk-8-later-jdk-releases.html#GUID-7BB28E4D-99B3-4078-BDC4-FC24180CE82B
+ *
+ * While we can handle this JVM quirk at build time, at runtime we won't know
+ * a user's JVM version. We'll provide documentation and an extension setting
+ * to add these flags to the extension-launched debugger backend.
+ */
+lazy val extraJvmOptions: Seq[String] =
+  if (scala.util.Properties.isJavaAtLeast("17"))
+    Seq(
+      "--add-opens",
+      "java.base/java.lang=ALL-UNNAMED",
+    )
+  else Seq()
+
+lazy val xjcSettings =
+  Seq(
+    libraryDependencies ++= Seq(
+      "com.sun.xml.bind" % "jaxb-impl" % "2.2.11",
+      "javax.activation" % "activation" % "1.1.1",
+      "org.glassfish.jaxb" % "jaxb-xjc" % "2.2.11",
+    ),
+    xjcCommandLine += "-nv",
+    xjcCommandLine += "-p",
+    xjcCommandLine += "org.apache.daffodil.tdml",
```

Comment:
Can we spend some time trying to figure out the reproducible builds issue with jaxb generated files that we saw with vscode? Daffodil currently has reproducible builds for jars (the other zip/tar/etc release artifacts have timestamp issues) and I would like to keep it that way. There's some discussions in the ASF about automating things like release signing, but that requires reproducible builds.

I see there's a `-no-header` xjc option that prevents a header comment containing a timestamp from being added. I wouldn't expect a comment to affect the .class file, so I'm not optimistic it would fix it, but at least it's one less piece of randomness to consider.

Reply:
It looks like the `-no-header` flag solves the issue for the .class and .jar files, but I'm not exactly sure why... I did confirm that all that flag does is strip out a comment header at the beginning of the file, so I don't know why that matters for .class files... I did confirm that the build is not reproducible without the `-no-header` flag. I also found out that there is a .class decompiler in VSCode (I think it might even be native). I decompiled both versions of a class that the `diff` command reported as different and did not see a change.

That being said, even with the `-no-header` flag, I notice differences in the generated .java files. These differences are the order of {@link ...} comments, so they also do not affect the actual code... Completely unsure as to why these comments don't matter and the header's do. The list of files that differ also varies, at least slightly, with every build.

If you can confirm that the current code on this branch creates a reproducible build for you, we'll leave `-no-header` in. If/When we attempt to switch over to scalaxb, we'll verify that we have a reproducible build.

Reply:
I am actually not able to reproduce the non-reproducable-build issue, either with or without the `-no-header` flag. So I'm not convinced that was the actual cause (which I think makes sense--I think we agree that comments shouldn't affect the class file).

I do see the different header dates and sometimes different ordering of `@links`, but none of that seems to cause the class files to be different for me. Seems like there must be something else going on. I can't reproduce it so I'm not sure what the cause would be...

The fact that sometimes `@links` show up in a different order or in different files makes me feel like maybe there is some source of randomness or ordering that *sometimes* does cause a change to the class file. Maybe no-header fixes that as a side-effect or maybe it's just not always reproducible, like I'm having trouble with now?